src/command/dev/schema.rs

```diff
@@ -149,11 +145,32 @@ impl SupergraphOpts {
                         polling_interval,
                         introspection_headers,
                     ),
-                    SchemaSource::Sdl { .. } | SchemaSource::Subgraph { .. } => {
-                        Err(RoverError::new(anyhow!(
-                            "Detected an invalid `graphref` or `sdl` schema source in {file}. rover dev only supports sourcing schemas via introspection and schema files. see https://www.apollographql.com/docs/rover/commands/supergraphs/#yaml-configuration-file for more information.",
-                            file = config_path.to_string()
-                        )))
+                    SchemaSource::Sdl { sdl } => {
+                        let routing_url = routing_url.ok_or_else(|| {
+                            anyhow!("`routing_url` must be set when providing SDL directly")
+                        })?;
+                        SubgraphSchemaWatcher::new_from_sdl(
+                            (name, routing_url),
+                            sdl,
+                            follower_messenger.clone(),
+                        )
+                    }
+                    SchemaSource::Subgraph { graphref, subgraph } => {
+                        let studio_client = if let Some(studio_client) = studio_client.as_ref() {
+                            studio_client
+                        } else {
+                            let client = client_config.get_authenticated_client(profile_opt)?;
+                            studio_client = Some(client);
+                            studio_client.as_ref().unwrap()
+                        };
+
+                        SubgraphSchemaWatcher::new_from_graph_ref(
+                            &graphref,
+                            routing_url,
+                            subgraph,
```

Comment:
This looks like the wrong subgraph name to me. 

When starting `rover dev --supergraph-config supergraph.yaml` with the following config:

```yaml
federation_version: =2.4.3
subgraphs:
  products:
    routing_url: 'http://localhost:4002'
    schema:
      file: './products/products.graphql'
  users:
    routing_url: 'http://localhost:4001'
    schema:
      file: './users/users.graphql'
  pandas:
    routing_url: 'http://localhost:4003'
    schema:
      file: './pandas/pandas.graphql'
  countries:
    schema:
      graphref: avery-fed@current
      subgraph: products
```

you get this in your terminal:

```console
$ rover dev --supergraph-config supergraph.yaml
⚠️  Do not run this command in production! ⚠️  It is intended for local development.
🛫 starting a session with the 'products' subgraph
🛫 starting a session with the 'users' subgraph
🛫 starting a session with the 'pandas' subgraph
🛫 starting a session with the 'products' subgraph
```

That last session should say `countries` instead of `products`, which it got from the `schema.graphref.subgraph` property in the YAML file rather than the `subgraph.countries` property.